### ISSUE TITLE
Npm install fails with latest node-rdkafka & node <= v10

### ISSUE THREAD

**Issue by kitd** [Blizzard/node-rdkafka#806]
Using Node v10.0, I get the following error when running an `npm install` on a project with node-rdkafka@2.9.0:

```
make[2]: Leaving directory `/home/kit/dev/noderdkafkatest/node_modules/node-rdkafka/deps/librdkafka/examples'
Updating CONFIGURATION.md
CONFIGURATION.md CONFIGURATION.md.tmp differ: byte 418, line 6
Installing documentation to /home/kit/dev/noderdkafkatest/node_modules/node-rdkafka/build/deps
install -d $DESTDIR/home/kit/dev/noderdkafkatest/node_modules/node-rdkafka/build/deps/share/doc/librdkafka
install LICENSE LICENSES.txt INTRODUCTION.md README.md CONFIGURATION.md STATISTICS.md $DESTDIR/home/kit/dev/noderdkafkatest/node_modules/node-rdkafka/build/deps/share/doc/librdkafka
make[1]: Leaving directory `/home/kit/dev/noderdkafkatest/node_modules/node-rdkafka/deps/librdkafka'
  TOUCH Release/obj.target/deps/librdkafka.stamp
  CXX(target) Release/obj.target/node-librdkafka/src/binding.o
../src/binding.cc: In function ‘void Init(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’:
../src/binding.cc:81:28: error: ‘GetCurrentEnvironment’ is not a member of ‘node’
   node::Environment* env = node::GetCurrentEnvironment(context);
                            ^
make: *** [Release/obj.target/node-librdkafka/src/binding.o] Error 1
rm deps_librdkafka_gyp_librdkafka_target_build_dependencies.intermediate
make: Leaving directory `/home/kit/dev/noderdkafkatest/node_modules/node-rdkafka/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/kit/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
gyp ERR! System Linux 3.10.0-1062.12.1.el7.x86_64
gyp ERR! command "/home/kit/.nvm/versions/node/v10.0.0/bin/node" "/home/kit/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/kit/dev/noderdkafkatest/node_modules/node-rdkafka
gyp ERR! node -v v10.0.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm WARN noderdkafkatest@1.0.0 No description
npm WARN noderdkafkatest@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-rdkafka@2.9.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-rdkafka@2.9.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kit/.npm/_logs/2020-06-16T13_03_15_567Z-debug.log
```
The debug log didn't seem to contain any info on the error.

**Environment Information**
 - OS [e.g. Mac, Arch, Windows 10]: RHEL 7.7 (Maipo) amd64 (also seen on MacOS)
 - Node Version [e.g. 8.2.1]: v10.0 & v11.0 
 - NPM Version [e.g. 5.4.2]: 6.4.1
 - C++ Toolchain [e.g. Visual Studio, llvm, g++]: g++
 - node-rdkafka version [e.g. 2.3.3]: 2.9.0

**Steps to Reproduce**

1. Set node version to v10.0
2. Create a simple node project with node-rdkafka@2.9.0 as a dependency
3. Run `npm install`. It fails with the above error.
4. Set node version to v11.0
5 Rerun `npm install`. It completes.

**node-rdkafka Configuration Settings**
Default.

**Additional context**
`nvm` was used to change the node versions
Also fails on node v.9.0

**Comment by iradul** [comment]
I see this happening with Node 9 however LTS versions (including 10) are working fine.
`docker run -ti --rm -w /test node:10 npm i node-rdkafka`

**Comment by kitd** [comment]
@iradul  Thx, yes I've just tried it on earlier LTSes. 4.91 & 8.17.0 both work fine.

Maybe this needs to be clarified in the docs?

**Comment by iradul** [comment]
I think we can easily fix this for Node 9.

**Comment by iradul** [comment]
Node 10, less then `10.16.0` also doesn't compile (#807).
I've tested `11.0.0` and `12.0.0`, both looking good.

**Comment by dlipeles** [comment]
Hi @iradul I'm currently seeing this error with node `11.0.0` 

```
'GetCurrentEnvironment' is not a member of 'node'
   node::Environment* env = node::GetCurrentEnvironment(context);
```

**Comment by iradul** [comment]
`docker run --rm -ti node:11.0 bash -c 'echo "Node version $(node -v)" && npm i node-rdkafka'`

**Comment by kitd** [comment]
@iradul That works for me, thanks.

**Comment by RicardoGralhoz** [comment]
@iradul is there a target node-rdkafka version for node **10.13.0 LTS**? 

I'm also having issues on `npm install node-rdkafka`; the most informative message after hanging says:
`deps/librdkafka.target.mk:13: recipe for target 'deps/librdkafka/config.h' failed`



#### Environment 
OS: Ubuntu 18.04.2 LTS
Node:  v10.13.0 LTS
NPM Version : 6.4.1
C++ Toolchain : g++
node-rdkafka version: 2.9.0

#### Command
```
cd /home/auser/versions/app/node_modules
npm install node-rdkafka --loglevel="verbose"
```

#### Output
```
npm info it worked if it ends with ok
npm verb cli [ '/home/auser/versions/node-v10.13.0/bin/node',
npm verb cli   '/home/auser/node/bin/npm',
npm verb cli   'install',
npm verb cli   'node-rdkafka',
npm verb cli   '--loglevel=verbose' ]
npm info using npm@6.4.1
npm info using node@v10.13.0
npm verb npm-session c1ea12ed
npm http fetch GET 304 https://registry.npmjs.org/node-rdkafka 126ms (from cache)
npm timing stage:loadCurrentTree Completed in 882ms
npm timing stage:loadIdealTree:cloneCurrentTree Completed in 3ms
npm timing stage:loadIdealTree:loadShrinkwrap Completed in 118ms
npm http fetch GET 304 https://registry.npmjs.org/nan 46ms (from cache)
npm http fetch GET 304 https://registry.npmjs.org/bindings 238ms (from cache)
npm http fetch GET 304 https://registry.npmjs.org/file-uri-to-path 55ms (from cache)
npm timing stage:loadIdealTree:loadAllDepsIntoIdealTree Completed in 510ms
npm timing stage:loadIdealTree Completed in 711ms
npm timing stage:generateActionsToTake Completed in 23ms
npm verb correctMkdir /home/auser/.npm/_locks correctMkdir not in flight; initializing
npm verb lock using /home/auser/.npm/_locks/staging-bf8cf57ec5c55454.lock for /home/auser/versions/app/node_modules/.staging
npm timing audit compress Completed in 8ms
npm info audit Submitting payload of 22296bytes
npm timing action:extract Completed in 214ms
npm info lifecycle undefined~preuninstall: undefined
npm info lifecycle undefined~uninstall: undefined
npm timing action:unbuild Completed in 2ms
npm timing action:remove Completed in 0ms
npm timing action:finalize Completed in 5ms
npm timing action:refresh-package-json Completed in 68ms
npm info lifecycle file-uri-to-path@1.0.0~preinstall: file-uri-to-path@1.0.0
npm info lifecycle bindings@1.5.0~preinstall: bindings@1.5.0
npm info lifecycle nan@2.14.1~preinstall: nan@2.14.1
npm info lifecycle node-rdkafka@2.9.0~preinstall: node-rdkafka@2.9.0
npm timing action:preinstall Completed in 2ms
npm info linkStuff file-uri-to-path@1.0.0
npm info linkStuff bindings@1.5.0
npm info linkStuff nan@2.14.1
npm info linkStuff node-rdkafka@2.9.0
npm timing action:build Completed in 2ms
npm info lifecycle file-uri-to-path@1.0.0~install: file-uri-to-path@1.0.0
npm info lifecycle bindings@1.5.0~install: bindings@1.5.0
npm info lifecycle nan@2.14.1~install: nan@2.14.1
npm info lifecycle node-rdkafka@2.9.0~install: node-rdkafka@2.9.0

> node-rdkafka@2.9.0 install /home/auser/versions/app/node_modules/node-rdkafka
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp verb cli [ '/home/auser/versions/node-v10.13.0/bin/node',
gyp verb cli   '/home/auser/versions/node-v10.13.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild' ]
gyp info using node-gyp@3.8.0
gyp info using node@10.13.0 | linux | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` succeeded python2 /usr/bin/python2
gyp verb check python version `/usr/bin/python2 -c "import sys; print "2.7.15
gyp verb check python version .%s.%s" % sys.version_info[:3];"` returned: %j
gyp verb get node dir no --target version specified, falling back to host node version: 10.13.0
gyp verb command install [ '10.13.0' ]
gyp verb install input version string "10.13.0"
gyp verb install installing version: 10.13.0
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 10.13.0
gyp verb build dir attempting to create "build" dir: /home/auser/versions/app/node_modules/node-rdkafka/build
gyp verb build dir "build" dir needed to be created? /home/auser/versions/app/node_modules/node-rdkafka/build
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: /home/auser/versions/app/node_modules/node-rdkafka/build/config.gypi
gyp verb config.gypi checking for gypi file: /home/auser/versions/app/node_modules/node-rdkafka/config.gypi
gyp verb common.gypi checking for gypi file: /home/auser/versions/app/node_modules/node-rdkafka/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/home/auser/versions/node-v10.13.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/auser/versions/app/node_modules/node-rdkafka/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/auser/versions/node-v10.13.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/auser/.node-gyp/10.13.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/auser/.node-gyp/10.13.0',
gyp info spawn args   '-Dnode_gyp_dir=/home/auser/versions/node-v10.13.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/auser/.node-gyp/10.13.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/auser/versions/app/node_modules/node-rdkafka',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp verb command build []
gyp verb build type Release
gyp verb architecture x64
gyp verb node dev dir /home/auser/.node-gyp/10.13.0
gyp verb `which` succeeded for `make` /usr/bin/make
gyp info spawn make
gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/auser/versions/app/node_modules/node-rdkafka/build'
  LD_LIBRARY_PATH=/home/auser/versions/app/node_modules/node-rdkafka/build/Release/lib.host:/home/auser/versions/app/node_modules/node-rdkafka/build/Release/lib.target:$LD_LIBRARY_PATH; export LD_LIBRARY_PATH; cd ../deps; mkdir -p librdkafka; node ../util/configure
checking for OS or distribution... ok (ubuntu)
checking for C compiler from CC env... failed
checking for gcc (by command)... ok
checking for C++ compiler from CXX env... failed
checking for C++ compiler (g++)... ok
checking executable ld... ok
checking executable nm... ok
checking executable objdump... ok
checking executable strip... ok
checking executable libtool... failed (disable)
checking for pkgconfig (by command)... failed
checking for install (by command)...
```
_**it hangs here; then pressing ^C**_
```
^Cdeps/librdkafka.target.mk:13: recipe for target 'deps/librdkafka/config.h' failed
make: *** [deps/librdkafka/config.h] Interrupt
npm verb lifecycle node-rdkafka@2.9.0~install: unsafe-perm in lifecycle true
npm verb lifecycle node-rdkafka@2.9.0~install: PATH: /home/auser/versions/node-v10.13.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/auser/versions/app/node_modules/node-rdkafka/node_modules/.bin:/home/auser/versions/app/node_modules/.bin:/home/auser/node/bin:/home/auser/bin:/home/auser/app/utils:/home/auser/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
npm verb lifecycle node-rdkafka@2.9.0~install: CWD: /home/auser/versions/app/node_modules/node-rdkafka
```
I can provide more info if needed.

**Comment by RicardoGralhoz** [comment]
The same issue also happens on trying to install node-rdkafka with different versions or on a different folder. This may have a different cause, so I opened issue #825

**Comment by RicardoGralhoz** [comment]
@iradul what is the latest version that is supported on Node 10, less than 10.16.0? I'm running node v10.13.0 LTS and can confirm I also got this issue.

**Comment by RicardoGralhoz** [comment]
> @iradul what is the latest version that is supported on Node 10, less than 10.16.0? I'm running node v10.13.0 LTS and can confirm I also got this issue.

It seems like at node-rdkafka
- v2.7.4  it worked
- v2.8.0  it broke node 8, 9 & 10-10.15 installation: 
  - 22452f8, Apr 5 2020, Support Node.js 13.x, #759
- v2.9.0  it fixes node 8 install
- _<no release as of Jul 22, 2020>_  it should fix 9 - 10.15
  - 0a77510 fixes node 9 installation
  - 41e7333 fixes node 10-10.15 installation

_I should update here later with more info..._

**Comment by iradul** [comment]
I published a new version (`v2.9.1`) that should fix this.
